lib/streamlit/runtime/state/session_state.py

```diff
@@ -298,6 +299,9 @@ class SessionState:
     # Keys used for widgets will be eagerly converted to the matching widget id
     _key_id_mapping: dict[str, str] = field(default_factory=dict)
 
+    # query params are stored in session state because query params will be tied with widget state at one point.
+    _query_params: QueryParams = field(default_factory=QueryParams)
```

Comment:
I think we can add it here for now and then the proxy can access it through session state. 

The other thought is to not have it in `session_state` and from what I can see, it's not too much to put it back in `session_state` when writing the `QueryParamsProxy` code.

Reply:
> The other side of the coin is to not have it in session_state and from what I can see, it's not too much work to put it back in session_state when writing the QueryParamsProxy code.

I don't understand what this means. Where would a session's `QueryParams` live if not in session state?